runtime/src/precompiles.rs

```diff
@@ -51,3 +57,36 @@ where
 fn hash(a: u64) -> H160 {
     H160::from_low_u64_be(a)
 }
+
+struct DispatchCallFilter;
+
+impl DispatchValidateT<AccountId, RuntimeCall> for DispatchCallFilter {
+    fn validate_before_dispatch(
+        _origin: &AccountId,
+        call: &RuntimeCall,
+    ) -> Option<fp_evm::PrecompileFailure> {
+        let info = call.get_dispatch_info();
+
+        if matches!(
```

Comment:
IMO `match` case block is more readable form of `else if <cond>` (vs case `_ if <cond>`).

Reply:
did you mean match? while they are indeed more readable, it would require more nested blocks in this particular case.